### ISSUE TITLE
Build Issue on Mac OS

### ISSUE THREAD

**Issue by hzyjerry** [facebookresearch/Replica-Dataset#15]
When running `build.sh`, I encountered the following issue with Mac OS

```
Scanning dependencies of target ptex
make[2]: *** No rule to make target `../3rdparty/Pangolin/build/src/libpangolin.so', needed by `ReplicaSDK/libptex.dylib'.  Stop.
make[2]: *** Waiting for unfinished jobs....
[ 10%] Building CXX object ReplicaSDK/CMakeFiles/ptex.dir/ptex/EGL.cpp.o
[ 20%] Building CXX object ReplicaSDK/CMakeFiles/ptex.dir/ptex/PLYParser.cpp.o
[ 30%] Building CXX object ReplicaSDK/CMakeFiles/ptex.dir/ptex/PTexLib.cpp.o
clang: clang: errorerror: : unsupported option '-fopenmp'
unsupported option '-fopenmp'
make[2]: *** [ReplicaSDK/CMakeFiles/ptex.dir/ptex/PLYParser.cpp.o] Error 1
make[2]: *** [ReplicaSDK/CMakeFiles/ptex.dir/ptex/EGL.cpp.o] Error 1
clang: error: unsupported option '-fopenmp'
make[2]: *** [ReplicaSDK/CMakeFiles/ptex.dir/ptex/PTexLib.cpp.o] Error 1
make[1]: *** [ReplicaSDK/CMakeFiles/ptex.dir/all] Error 2
make: *** [all] Error 2
```

My gcc version is 9.1.0. Would appreciate any insight!

**Comment by mp3guy** [comment]
The build.sh script is meant for a linux distribution like Ubuntu or Fedora, we haven't made specific Mac OS support yet. In principal it should work with a few small changes.

**Comment by jstraub** [comment]
It looks like openmp might need to be installed? Currently we do not support Mac OS.

**Comment by jstraub** [comment]
I had to give cmake the right gcc (not the default Apple one but the one installed via brew)
```
cmake -DCMAKE_CXX_COMPILER=g++-9 ..
```

**Comment by bigbike** [comment]
> The build.sh script is meant for a linux distribution like Ubuntu or Fedora, we haven't made specific Mac OS support yet. In principal it should work with a few small changes.

Good to know.

The current README.md is a bit misleading in my opinion. Do you think it is necessary to add your comments to the README.md in order to avoid the guess work by the users?

Or if the changes are small to make it fly on Mac OS, can we improve the build.sh to make it happen for the users?

Thanks a lot!

**Comment by bigbike** [comment]
> It looks like openmp might need to be installed? Currently we do not support Mac OS.

Some thoughts:
If openmp is a must-have component, can we add it as a dependency to the SDK in the cmake?
e.g., find_package(OpenMP). Meanwhile create a script file to "brew install libomp" for the user (like the install_deps.sh in the Habitat Simulator).

By the way, can we also add the "pangolin" as a must-have dependency to the SDK? When it is missing, the cmake can automatically grab it from github and compile it for the user.

Thanks!

**Comment by bigbike** [comment]
> I had to give cmake the right gcc (not the default Apple one but the one installed via brew)
> 
> ```
> cmake -DCMAKE_CXX_COMPILER=g++-9 ..
> ```

Good to know. Thanks a lot! Would you consider mentioning it in the README.md to help the MacOS users to avoid this "trap"?

By the way, it will complain <EGL/egl.h> is missing when compiling the Replica-Dataset/ReplicaSDK/ptex/EGL.cpp.

**Comment by bigbike** [comment]
By the way, we looked into the "shader" code, which required OpenGL 4.3. However, the highest version on MacOS is just 4.1.
Even if we solve all the compiling issues on mac, I suspect it would still not render the model. 
A bit out of topic. But just mention it here.

**Comment by jstraub** [comment]
Yeah it seems that for now Mac OS is not supported unfortunately.

**Comment by johnwlambert** [comment]
@jstraub Hi Julian, thanks very much for the dataset. Might be good to add `cmake` as a required dependency on the readme, I didn't see it listed there.

**Comment by amir32002** [comment]
+1 on this one. I resolved it by guesswork, but it could be smoother. Also OpenGL 4.3 is not supported on any Mac model as of now https://support.apple.com/en-us/HT202823
